<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Cooperative Estimation with AFT Survival Models — aft_coop • AFTCoop</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Cooperative Estimation with AFT Survival Models — aft_coop"><meta property="og:description" content="The function aft_coop implements a cooperative estimation method for Accelerated Failure Time (AFT) models
allowing to integrate two matrices U and Z of high-dimensional covariates.
The two matrices  U and Z can represent two omics views, such as gene expression and methylation.
When used with two matrices  U and Z, the function estimates the coefficients \(\beta_u\) and \(\beta_z\)
associated with the two views by solving the minimization problem using the proximal gradient descend algorithm.
In such a case, the minimization problem consists of three terms: the negative log-likelihood, the agreement term among the two views,
and the Lasso penalization for each view.
However, the function works also with a single matrix  U or Z.
In such cases, it performs negative log-likelihood minimization with the Lasso penalty.
In both cases, cross-validation estimates the regularization parameter.
The function supports parallel processing and cross-validation for optimal parameter selection."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">AFTCoop</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Cooperative Estimation with AFT Survival Models</h1>

    <div class="hidden name"><code>aft_coop.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The function aft_coop implements a cooperative estimation method for Accelerated Failure Time (AFT) models
allowing to integrate two matrices <code>U</code> and <code>Z</code> of high-dimensional covariates.
The two matrices  <code>U</code> and <code>Z</code> can represent two omics views, such as gene expression and methylation.
When used with two matrices  <code>U</code> and <code>Z</code>, the function estimates the coefficients \(\beta_u\) and \(\beta_z\)
associated with the two views by solving the minimization problem using the proximal gradient descend algorithm.
In such a case, the minimization problem consists of three terms: the negative log-likelihood, the agreement term among the two views,
and the Lasso penalization for each view.
However, the function works also with a single matrix  <code>U</code> or <code>Z</code>.
In such cases, it performs negative log-likelihood minimization with the Lasso penalty.
In both cases, cross-validation estimates the regularization parameter.
The function supports parallel processing and cross-validation for optimal parameter selection.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">aft_coop</span><span class="op">(</span></span>
<span>  <span class="va">U</span>,</span>
<span>  <span class="va">Z</span>,</span>
<span>  <span class="va">Y</span>,</span>
<span>  <span class="va">delta</span>,</span>
<span>  <span class="va">sigma</span>,</span>
<span>  nfolds <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  <span class="va">model</span>,</span>
<span>  <span class="va">case</span>,</span>
<span>  <span class="va">rho_values</span>,</span>
<span>  <span class="va">lam_min</span>,</span>
<span>  nlambda <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  lambda.ratio.min <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  parallel <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  ncore_max_rho <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  ncore_max_cv <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">123</span>,</span>
<span>  iplot <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  addbar <span class="op">=</span> <span class="cn">F</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-u">U<a class="anchor" aria-label="anchor" href="#arg-u"></a></dt>
<dd><p>Matrix of covariates correspondig to the first view (e.g., gene expression) of dimension <code>n x p_u</code>, where <code>n</code> is the number of samples and <code>p_u</code> is the number of variables in the first view.</p></dd>


<dt id="arg-z">Z<a class="anchor" aria-label="anchor" href="#arg-z"></a></dt>
<dd><p>Matrix of covariates correspondig to the second view (e.g., gene methylation) of dimension <code>n x p_z</code>,where <code>n</code> is the number of samples and <code>p_z</code> is the number of variables in the second view.</p></dd>


<dt id="arg-y">Y<a class="anchor" aria-label="anchor" href="#arg-y"></a></dt>
<dd><p>Vector of log observed survival times or log censored times (i.e., the response variables are in log scale) of dimension <code>n</code>, where <code>n</code> is the number of samples.</p></dd>


<dt id="arg-delta">delta<a class="anchor" aria-label="anchor" href="#arg-delta"></a></dt>
<dd><p>Vector of censoring indicators (1 for event, 0 for censored) of dimension  <code>n</code>, where <code>n</code> is the number of samples.</p></dd>


<dt id="arg-sigma">sigma<a class="anchor" aria-label="anchor" href="#arg-sigma"></a></dt>
<dd><p>Scale parameter for the AFT model (it must be a positive real value).</p></dd>


<dt id="arg-nfolds">nfolds<a class="anchor" aria-label="anchor" href="#arg-nfolds"></a></dt>
<dd><p>Number of folds to use for the cross-validation (default: 5)</p></dd>


<dt id="arg-model">model<a class="anchor" aria-label="anchor" href="#arg-model"></a></dt>
<dd><p>Character string specifying the specific AFT model to use.  It can assume one of the following values "weibull";"lognormal";"loglogistic".
The noise distribution in the log time models  \(Y=X \beta+\sigma \epsilon\) can be Extreme Value Gumbel distribution that corresponds to the "weibull" model, normal distribution that correspond to the  "lognormal" model, and logistic distribution that correspond to the "loglogistic" model.</p></dd>


<dt id="arg-case">case<a class="anchor" aria-label="anchor" href="#arg-case"></a></dt>
<dd><p>Character string specifying the data model. It can take the following values: "onlyU","onlyZ","coop". The choice "onlyU" uses only the first view of covariates,
The choice "onlyZ" uses only the second view of covariates, The choice "coop" uses both views as covariates.</p></dd>


<dt id="arg-rho-values">rho_values<a class="anchor" aria-label="anchor" href="#arg-rho-values"></a></dt>
<dd><p>Vector of <code>nrhos</code> cooperation parameters. The  <code>nrhos</code> parameters are relevant only when case="coop" and allow different tuning of the coperative and Lasso penalization terms. When case="onlyU" or case="onlyZ", the parameter is not relevant and it is set to 1.</p></dd>


<dt id="arg-lam-min">lam_min<a class="anchor" aria-label="anchor" href="#arg-lam-min"></a></dt>
<dd><p>Logical; if TRUE uses lambda.min, if FALSE uses lambda.1se</p></dd>


<dt id="arg-nlambda">nlambda<a class="anchor" aria-label="anchor" href="#arg-nlambda"></a></dt>
<dd><p>Number of lambda values in regularization path (default: 100)</p></dd>


<dt id="arg-lambda-ratio-min">lambda.ratio.min<a class="anchor" aria-label="anchor" href="#arg-lambda-ratio-min"></a></dt>
<dd><p>Minimum ratio of smallest to largest lambda (default: 0.01)</p></dd>


<dt id="arg-parallel">parallel<a class="anchor" aria-label="anchor" href="#arg-parallel"></a></dt>
<dd><p>Logical; enable parallel processing (default: TRUE)</p></dd>


<dt id="arg-ncore-max-rho">ncore_max_rho<a class="anchor" aria-label="anchor" href="#arg-ncore-max-rho"></a></dt>
<dd><p>Maximum number of cores for parallel processing for splitting over rho_values (default: 4)</p></dd>


<dt id="arg-ncore-max-cv">ncore_max_cv<a class="anchor" aria-label="anchor" href="#arg-ncore-max-cv"></a></dt>
<dd><p>Maximum number of cores for parallel processing for splitting over Cross-validation (default: 5)</p></dd>


<dt id="arg-seed">seed<a class="anchor" aria-label="anchor" href="#arg-seed"></a></dt>
<dd><p>Random seed for reproducibility (default: 123)</p></dd>


<dt id="arg-iplot">iplot<a class="anchor" aria-label="anchor" href="#arg-iplot"></a></dt>
<dd><p>Logical; enable the plot of the CV function (default: TRUE)</p></dd>


<dt id="arg-addbar">addbar<a class="anchor" aria-label="anchor" href="#arg-addbar"></a></dt>
<dd><p>Logical; when iplot is true, add the error bar to the plot (default: FALSE)</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A matrix of dimension <code>(p_u+p_z)x nrhos</code>,where:</p><ul><li><p>Rows correspond to the estimates regression coefficients \(\hat{\beta}=[\hat{\beta}_u,\hat{\beta}_z]\) for each view.</p></li>
<li><p>Columns correspond to the estimates for different rho values</p></li>
<li><p>Values are coefficient \(\hat{\beta}\) in original scale.
The first set of coefficients corresponds \(\hat{\beta}_u\) to those associated with the <code>U</code> view, the second set \(\hat{\beta_z}\) to those associated to the view  <code>Z</code>.</p></li>
</ul></div>
    <div id="details">
    <h2>Details</h2>
    <p>When providing two matrices <code>U</code> and <code>Z</code>, the function aft_coop performs the following steps:</p><ul><li><p>Build combined matrix <code>X=[U;Z]</code> and <code>Xtilde=[U;-Z]</code>, given the two matrices <code>U</code> and <code>Z</code></p></li>
<li><p>Standardizes <code>U</code> and <code>Z</code>  matrices.</p></li>
<li><p>Generates lambda grid and initialize the search using gradient at initial estimate \(\beta = 0\).</p></li>
<li><p>For each rho value:</p><ul><li><p>Performs K-fold cross-validation to select optimal lambda</p></li>
<li><p>Computes cooperative matrix</p></li>
<li><p>Applies proximal gradient descent</p></li>
<li><p>Rescales coefficients back to original scale</p></li>
</ul></li>
</ul><p>The function uses standardized variables internally but returns coefficients
in the original scale. Cross-validation is used to select optimal lambda values.
The user can choose the optimal value for lambda as lambda.min (i.e., the values of lambda for which the CV is minimised)
or lambda.1se (i.e, the largests value of lambda within 1 standard error form the the minumum).
The latter choice provides a sparsest model.
The choice of the type of optimal value to use is done setting the logical lam_min parameter.
However, we switch between  lambda.min  and lambda.1se is the correspondoing values is at the boarder of the grid interval.
Parallel processing implementation differs between Windows (using makeCluster)
and Unix-like systems (using mclapply).</p>
    </div>
    <div id="note">
    <h2>Note</h2>
    <p>Last change 15/04/2025</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Angelini, De Canditiis, De Feis, Iuliano (in prep. 2025).</p>
<p>Ding, Li, Narasimhan, Tibshirani, "Cooperative learning for multiview analysis", PNAS (2022)
DOI: 10.1073/pnas.220211311
Available at: <a href="https://www.pnas.org/doi/abs/10.1073/pnas.2202113119" class="external-link">https://www.pnas.org/doi/abs/10.1073/pnas.2202113119</a></p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="cv.coop.html">cv.coop</a></code> for underlying implementation details</p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Generate example data</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">model</span><span class="op">=</span><span class="st">"weibull"</span></span></span>
<span class="r-in"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="generate_data.html">generate_data</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  model <span class="op">=</span> <span class="va">model</span>,</span></span>
<span class="r-in"><span>    n <span class="op">=</span> <span class="fl">200</span>,</span></span>
<span class="r-in"><span>    pu <span class="op">=</span> <span class="fl">150</span>,</span></span>
<span class="r-in"><span>    pz <span class="op">=</span> <span class="fl">150</span>,</span></span>
<span class="r-in"><span>    tu <span class="op">=</span> <span class="fl">6</span>,</span></span>
<span class="r-in"><span>    tz <span class="op">=</span> <span class="fl">6</span>,</span></span>
<span class="r-in"><span>    rate <span class="op">=</span> <span class="fl">40</span>,</span></span>
<span class="r-in"><span>    sigma_true <span class="op">=</span> <span class="fl">0.5</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>    <span class="va">Y</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">$</span><span class="va">Y</span>  <span class="co"># vector of log survival times or censored times</span></span></span>
<span class="r-in"><span>    <span class="va">delta</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">$</span><span class="va">delta</span>  <span class="co"># vector of censoring indicators</span></span></span>
<span class="r-in"><span>    <span class="va">U</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">$</span><span class="va">U</span>  <span class="co"># matrix corresponding to the first view</span></span></span>
<span class="r-in"><span>    <span class="va">Z</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">$</span><span class="va">Z</span>  <span class="co"># matrix corresponding to the second view</span></span></span>
<span class="r-in"><span>    <span class="va">fit_survreg</span> <span class="op">&lt;-</span> <span class="fu">survreg</span><span class="op">(</span><span class="fu">Surv</span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">times_c</span>, <span class="va">delta</span><span class="op">)</span><span class="op">~</span><span class="fl">1</span>, dist<span class="op">=</span><span class="va">model</span>,scale<span class="op">=</span><span class="fl">0</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">sigma.est</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">exp</a></span><span class="op">(</span><span class="va">fit_survreg</span><span class="op">$</span><span class="va">icoef</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">rho_values</span><span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">0.25</span>,<span class="fl">0.5</span>,<span class="fl">0.75</span><span class="op">)</span> <span class="co">#vector parameteters for rho values</span></span></span>
<span class="r-in"><span><span class="co"># Using aft_coop with two views</span></span></span>
<span class="r-in"><span>    <span class="va">beta_est_coop</span><span class="op">&lt;-</span><span class="fu">aft_coop</span><span class="op">(</span></span></span>
<span class="r-in"><span>    U<span class="op">=</span><span class="va">U</span>,</span></span>
<span class="r-in"><span>    Z<span class="op">=</span><span class="va">Z</span>,</span></span>
<span class="r-in"><span>    Y<span class="op">=</span><span class="va">Y</span>,</span></span>
<span class="r-in"><span>    delta<span class="op">=</span><span class="va">delta</span>,</span></span>
<span class="r-in"><span>    sigma<span class="op">=</span><span class="va">sigma.est</span>,</span></span>
<span class="r-in"><span>    nfolds<span class="op">=</span><span class="fl">5</span>,</span></span>
<span class="r-in"><span>    model<span class="op">=</span><span class="va">model</span>,</span></span>
<span class="r-in"><span>    case<span class="op">=</span><span class="st">"coop"</span>,</span></span>
<span class="r-in"><span>    rho_values<span class="op">=</span>  <span class="va">rho_values</span>,</span></span>
<span class="r-in"><span>    lam_min<span class="op">=</span><span class="cn">F</span>,</span></span>
<span class="r-in"><span>    parallel<span class="op">=</span><span class="cn">T</span>,</span></span>
<span class="r-in"><span>    ncore_max<span class="op">=</span><span class="fl">5</span>,</span></span>
<span class="r-in"><span>    seed<span class="op">=</span><span class="fl">123</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>   <span class="co">## estimate of beta  (the first pu components refers to the view U; The second pz component to view Z  )</span></span></span>
<span class="r-in"><span>   <span class="va">beta_est_coop</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Using aft_coop with only the first view U</span></span></span>
<span class="r-in"><span>    <span class="va">beta_est_onlyU</span><span class="op">&lt;-</span><span class="fu">aft_coop</span><span class="op">(</span></span></span>
<span class="r-in"><span>    U<span class="op">=</span><span class="va">U</span>,</span></span>
<span class="r-in"><span>    Z<span class="op">=</span><span class="va">Z</span>,</span></span>
<span class="r-in"><span>    Y<span class="op">=</span><span class="va">Y</span>,</span></span>
<span class="r-in"><span>    delta<span class="op">=</span><span class="va">delta</span>,</span></span>
<span class="r-in"><span>    sigma<span class="op">=</span><span class="va">sigma.est</span>,</span></span>
<span class="r-in"><span>    nfolds<span class="op">=</span><span class="fl">5</span>,</span></span>
<span class="r-in"><span>    model<span class="op">=</span><span class="va">model</span>,</span></span>
<span class="r-in"><span>    case<span class="op">=</span><span class="st">"onlyU"</span>,</span></span>
<span class="r-in"><span>    rho_values<span class="op">=</span>  <span class="fl">1</span>,</span></span>
<span class="r-in"><span>    lam_min<span class="op">=</span><span class="cn">F</span>,</span></span>
<span class="r-in"><span>    parallel<span class="op">=</span><span class="cn">T</span>,</span></span>
<span class="r-in"><span>    ncore_max<span class="op">=</span><span class="fl">5</span>,</span></span>
<span class="r-in"><span>    seed<span class="op">=</span><span class="fl">123</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Using aft_coop with only the second view</span></span></span>
<span class="r-in"><span> <span class="va">beta_est_onlyZ</span><span class="op">&lt;-</span><span class="fu">aft_coop</span><span class="op">(</span></span></span>
<span class="r-in"><span>    U<span class="op">=</span><span class="va">U</span>,</span></span>
<span class="r-in"><span>    Z<span class="op">=</span><span class="va">Z</span>,</span></span>
<span class="r-in"><span>    Y<span class="op">=</span><span class="va">Y</span>,</span></span>
<span class="r-in"><span>    delta<span class="op">=</span><span class="va">delta</span>,</span></span>
<span class="r-in"><span>    sigma<span class="op">=</span><span class="va">sigma.est</span>,</span></span>
<span class="r-in"><span>    nfolds<span class="op">=</span><span class="fl">5</span>,</span></span>
<span class="r-in"><span>    model<span class="op">=</span><span class="va">model</span>,</span></span>
<span class="r-in"><span>    case<span class="op">=</span><span class="st">"onlyZ"</span>,</span></span>
<span class="r-in"><span>    rho_values<span class="op">=</span>  <span class="fl">1</span>,</span></span>
<span class="r-in"><span>    lam_min<span class="op">=</span><span class="cn">F</span>,</span></span>
<span class="r-in"><span>    parallel<span class="op">=</span><span class="cn">T</span>,</span></span>
<span class="r-in"><span>    ncore_max<span class="op">=</span><span class="fl">5</span>,</span></span>
<span class="r-in"><span>    seed<span class="op">=</span><span class="fl">123</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>    <span class="va">beta_est_onlyZ</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Claudia Angelini, Daniela De Canditiis, Italia De Feis, Antonella Iuliano.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

