<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Generate Simulated Survival Accelerated Failure Time model Data with Two Views U and Z. — generate_data • AFTCoop</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Generate Simulated Survival Accelerated Failure Time model Data with Two Views U and Z. — generate_data"><meta property="og:description" content="Function generate_data generates simulated survival data for Accelerated Failure Time (AFT) models with two views
here,  U and Z, that are correlated with unobserved latent variables (L).
The two observed views U and Z might represent two omics data matrices such as gene expression and methylation. The latent factor L can represent some unobserved factors that are associated with the survival times.
Some of the observed variables in  U and Z are correlated with the latent factors L.
The simulator allows to consider different proportion of right censoring.
The simulation scheme extends to the AFT models the simulation scheme proposed in Ding et al, 2022 for assessing the performance of cooperative regression,
and it is described in detail in Angelini et al 2025.
The function supports Weibull, Lognormal, and Loglogistic distributions for the error noise."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">AFTCoop</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate Simulated Survival Accelerated Failure Time model Data with Two Views U and Z.</h1>

    <div class="hidden name"><code>generate_data.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Function generate_data generates simulated survival data for Accelerated Failure Time (AFT) models with two views
here,  <code>U</code> and <code>Z</code>, that are correlated with unobserved latent variables (<code>L</code>).
The two observed views <code>U</code> and <code>Z</code> might represent two omics data matrices such as gene expression and methylation. The latent factor <code>L</code> can represent some unobserved factors that are associated with the survival times.
Some of the observed variables in  <code>U</code> and <code>Z</code> are correlated with the latent factors <code>L</code>.
The simulator allows to consider different proportion of right censoring.
The simulation scheme extends to the AFT models the simulation scheme proposed in Ding et al, 2022 for assessing the performance of cooperative regression,
and it is described in detail in Angelini et al 2025.
The function supports Weibull, Lognormal, and Loglogistic distributions for the error noise.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">generate_data</span><span class="op">(</span></span>
<span>  <span class="va">model</span>,</span>
<span>  <span class="va">n</span>,</span>
<span>  <span class="va">pu</span>,</span>
<span>  <span class="va">pz</span>,</span>
<span>  <span class="va">tu</span>,</span>
<span>  <span class="va">tz</span>,</span>
<span>  rate <span class="op">=</span> <span class="fl">40</span>,</span>
<span>  <span class="va">sigma_true</span>,</span>
<span>  rl <span class="op">=</span> <span class="fl">40</span>,</span>
<span>  rl_pos <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  snr <span class="op">=</span> <span class="fl">0.8</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-model">model<a class="anchor" aria-label="anchor" href="#arg-model"></a></dt>
<dd><p>Character string specifying the noise distribution. In can assume one of the following values: "weibull", "lognormal", or "loglogistic".</p></dd>


<dt id="arg-n">n<a class="anchor" aria-label="anchor" href="#arg-n"></a></dt>
<dd><p>Integer; number of observations (it must be a positive integer number)</p></dd>


<dt id="arg-pu">pu<a class="anchor" aria-label="anchor" href="#arg-pu"></a></dt>
<dd><p>Integer; number of covariates in view <code>U</code> of dimension  <code>n x p_u</code></p></dd>


<dt id="arg-pz">pz<a class="anchor" aria-label="anchor" href="#arg-pz"></a></dt>
<dd><p>Integer; number of covariates in view <code>Z</code> of dimension  <code>n x p_z</code></p></dd>


<dt id="arg-tu">tu<a class="anchor" aria-label="anchor" href="#arg-tu"></a></dt>
<dd><p>Numeric; strength of view <code>U</code> in relationship with latent variables <code>L</code>.</p></dd>


<dt id="arg-tz">tz<a class="anchor" aria-label="anchor" href="#arg-tz"></a></dt>
<dd><p>Numeric; strength of view <code>Z</code> in relationship with latent variables <code>L</code>.</p></dd>


<dt id="arg-rate">rate<a class="anchor" aria-label="anchor" href="#arg-rate"></a></dt>
<dd><p>Numeric; minimum right censoring rate for the observed survival times.</p></dd>


<dt id="arg-sigma-true">sigma_true<a class="anchor" aria-label="anchor" href="#arg-sigma-true"></a></dt>
<dd><p>Numeric; true scale parameter for the AFT model</p></dd>


<dt id="arg-rl">rl<a class="anchor" aria-label="anchor" href="#arg-rl"></a></dt>
<dd><p>Integer; number of relevant latent covariates (default: 40).</p></dd>


<dt id="arg-rl-pos">rl_pos<a class="anchor" aria-label="anchor" href="#arg-rl-pos"></a></dt>
<dd><p>Integer; number of positively associated latent covariates (default: 20)</p></dd>


<dt id="arg-snr">snr<a class="anchor" aria-label="anchor" href="#arg-snr"></a></dt>
<dd><p>Numeric; signal-to-noise ratio (default: 0.8)</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A list containing:</p><ul><li><p>Y: Log-transformed censored survival times</p></li>
<li><p>times_c: Original scale censored survival times</p></li>
<li><p>delta: Censoring indicators (1 = event, 0 = censored)</p></li>
<li><p>L: Matrix of latent variables</p></li>
<li><p>beta_L: True coefficients for latent variables</p></li>
<li><p>U: First view matrix</p></li>
<li><p>Z: Second view matrix</p></li>
</ul></div>
    <div id="details">
    <h2>Details</h2>
    <p>The data generation process follows these steps:</p><ul><li><p>Generates latent variables (L) from normal distribution</p></li>
<li><p>Creates views U and Z with controlled relationship to <code>L</code>, to this purpose <code>tu</code> and <code>tz</code> represent the correlation among the rl relevant variables in the latent factor <code>L</code> and the first rl variables in both <code>U</code> and <code>Z</code>.</p></li>
<li><p>Generates survival times based on specified distribution from the latent variables <code>L</code>.</p></li>
<li><p>Applies random censoring to achieve desired censoring rate.</p></li>
</ul><p>The function requires the flexsurv package for the loglogistic model.</p>
<p>The function uses the following parameter ranges:</p><ul><li><p>Positive coefficients: Uniform(0.1, 1)</p></li>
<li><p>Negative coefficients: Uniform(-1, -0.1)</p></li>
<li><p>Censoring: Exponential distribution</p></li>
</ul><p>View strengths (tu, tz) typically use these combinations:</p><ul><li><p>(6,6): Strong and equal effect in both views U and Z.</p></li>
<li><p>(6,4): Strong view in U, moderate effect in Z.</p></li>
<li><p>(6,2): Strong view in U, weak effect in Z.</p></li>
<li><p>(6,0): Strong view in U, No effect in Z.</p></li>
</ul><p>The choice of the SNR (default value snr=0.8) allow to generate different challenging scenarios.</p>
<p>The function prints the actual percentage of non-censored observations and
a summary of the censored survival times. If the variance constraints cannot
be satisfied with the given parameters, it will adjust the view variances
and print a warning message.</p>
    </div>
    <div id="note">
    <h2>Note</h2>
    <p>Last change 05/06/2025</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Angelini, De Canditiis, De Feis, Iuliano (in prep. 2025).</p>
<p>Ding, D., Li, S., Narasimhan, B., Tibshirani, R. (2021) <a href="doi:10.1073/pnas.2202113119">doi:10.1073/pnas.2202113119</a>)</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Generate Weibull survival data</span></span></span>
<span class="r-in"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu">generate_data</span><span class="op">(</span></span></span>
<span class="r-in"><span>  model <span class="op">=</span> <span class="st">"weibull"</span>,</span></span>
<span class="r-in"><span>  n <span class="op">=</span> <span class="fl">200</span>,</span></span>
<span class="r-in"><span>  pu <span class="op">=</span> <span class="fl">150</span>,</span></span>
<span class="r-in"><span>  pz <span class="op">=</span> <span class="fl">150</span>,</span></span>
<span class="r-in"><span>  tu <span class="op">=</span> <span class="fl">6</span>,</span></span>
<span class="r-in"><span>  tz <span class="op">=</span> <span class="fl">6</span>,</span></span>
<span class="r-in"><span>  rate <span class="op">=</span> <span class="fl">40</span>,</span></span>
<span class="r-in"><span>  sigma_true <span class="op">=</span> <span class="fl">0.5</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Access components</span></span></span>
<span class="r-in"><span><span class="va">Y</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">$</span><span class="va">Y</span>  <span class="co"># log survival times</span></span></span>
<span class="r-in"><span><span class="va">delta</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">$</span><span class="va">delta</span>  <span class="co"># censoring indicators</span></span></span>
<span class="r-in"><span><span class="va">U</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">$</span><span class="va">U</span>  <span class="co"># first view</span></span></span>
<span class="r-in"><span><span class="va">Z</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">$</span><span class="va">Z</span>  <span class="co"># second view</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Claudia Angelini, Daniela De Canditiis, Italia De Feis, Antonella Iuliano.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

